### DISCUSSION TITLE
NiceGui and NGINX Unit

### DISCUSSION THREAD

**Discussion by interblitz** [zauberzeug/nicegui#2117] [Q&A] [answered]
### Question

Is NiceGui fully compatible with [NGINX Unit](https://unit.nginx.org/)?

**Comment by rodja** [comment, marked as answer]
NiceGUI is fully compatible with FastAPI so it should work out of the box: https://unit.nginx.org/howto/fastapi/. Just make sure you restrict the number of process to one (see https://github.com/zauberzeug/nicegui/discussions/1539), because Unit does not support session affinity as far as I know.

**Reply by interblitz** [reply]
Great, thank you!
